pkg/controller/endpointslicemirroring/endpointslicemirroring_controller.go

```diff
@@ -488,12 +488,15 @@ func (c *Controller) onEndpointSliceDelete(obj interface{}) {
 		utilruntime.HandleError(fmt.Errorf("onEndpointSliceDelete() expected type discovery.EndpointSlice, got %T", obj))
 		return
 	}
-	if managedByController(endpointSlice) && c.endpointSliceTracker.Has(endpointSlice) {
-		// This returns false if we didn't expect the EndpointSlice to be
-		// deleted. If that is the case, we queue the Service for another sync.
-		if !c.endpointSliceTracker.HandleDeletion(endpointSlice) {
-			c.queueEndpointsForEndpointSlice(endpointSlice)
-		}
+	// if endpoint not managed by controller, just skip this EndpointSlice
+	if !managedByController(endpointSlice) {
+		return
+	}
+	// requeue the service for another sync, if
+	// 1. endpointSliceTracker don't has endpoint.
```

Comment:
why should not be requeued if it is managed and the `endpointSliceTracker` has the endpoint?
1. I create endpoint and is mirrored, is not added to the tracker here?
2. I delete the endpointslice, is in the tracker but I don't requeue so it is not recreated
I think I should recreate it

Reply:
sorry, my mistake, misspelled, endpointSliceTracker don't has this endpointslice not endpoint

Reply:
ah, no, I didn't mean the naming, I mean, what happens if the endpointSliceTracker has the the endpoint slice and is deleted? why don't we reconcile

Is not the problem that the slice is not in the tracker? should not handle this case and add it to the tracker?

@robscott have a better understanding of the internals of this controller

Reply:
under my understanding, tracker may avoid some race condition, if controller create endpointslice，but may not receive event from apiserver，and lister can not get this endpointslice from lister， and if we do a reconcile，may lead something wrong. so controller call tracker ExpectDeletion/Update function when it call Create/Update/Delete API, update new generation in tracker.

if we has stale information, will skip this sync, try later
``` go
if c.endpointSliceTracker.StaleSlices(svc, endpointSlices) {
		return endpointsliceutil.NewStaleInformerCache("EndpointSlice informer cache is out of date")
}
```